### DISCUSSION TITLE
Local emulator usage example

### DISCUSSION THREAD

**Discussion by quasi-coherent** [Byron/google-apis-rs#313] [Q&A] [answered]
Is it possible to use `google-pubsub1` with the [local emulator](https://cloud.google.com/pubsub/docs/emulator)?  Looking at any of the examples, it doesn't seem like one can construct a [PubSub](https://docs.rs/google-pubsub1/2.0.8+20210322/google_pubsub1/api/struct.Pubsub.html) to do this, but I might be missing something.  

I'd be happy to contribute some documentation if it is possible.

**Comment by Byron** [comment, marked as answer]
An interesting question that unfortunately I don't have an answer for. For all I know, these APIs are generated from the official google specification which typically means they do as advertised. However, there have been cases where certain ways of doing things as advertised in the google docs for such API weren't possible.

Maybe it's worth a try.

**Reply by quasi-coherent** [reply]
Thanks for the quick reply!

It's more that the PubSub client uses yup_oauth2, and I don't see a way to use no authentication with that library.  In the local emulator case, the PubSub runs on `localhost:<port>` with no authentication.  

In the Python SDK, it's possible to create a `PublishClient` without any credentials.

**Reply by Byron** [reply]
Happily, and I see. Indeed right now it will always try to authenticate, but ultimately it's just a token that is placed into the HTTP request. Now I wonder if the `yupoauth2` part is just an implementation of a trait that could also be implemented to yield just an empty string for a token or a bogus value.
If the local pubsub client would then ignore the token, it could work.
